db/model/Gdoc/upsertGdoc.test.ts

```diff
@@ -0,0 +1,148 @@
+import { describe, it, expect, afterEach, beforeAll, vi } from "vitest"
+import {
+    OwidGdocType,
+    EnrichedBlockText,
+    PostsGdocsTableName,
+    PostsGdocsComponentsTableName,
+    OwidGdocPublicationContext,
+} from "@ourworldindata/types"
+import * as Types from "@ourworldindata/utils"
+
+vi.mock("./GdocDataInsight.js", () => ({
+    GdocDataInsight: class {},
+}))
+vi.mock("./GdocHomepage.js", () => ({
+    GdocHomepage: class {},
+}))
+vi.mock("./GdocAbout.js", () => ({
+    GdocAbout: class {},
+}))
+vi.mock("./GdocAuthor.js", () => ({
+    GdocAuthor: class {},
+}))
+vi.mock("./GdocAnnouncement.js", () => ({
+    GdocAnnouncement: class {},
+}))
+vi.mock("./extractGdocComponentInfo.js", () => ({
+    getGdocComponentsWithoutChildren: (): never[] => [],
+}))
+
+// The tests in this file mock all the database calls and are only used to verify
+// that the markdown regeneration logic is correctly triggered before saving.
+// There is a separate integration test in adminSiteServer/tests/gdocs-markdown.test.ts
+// that verifies end-to-end that the markdown is correctly regenerated and saved.
+// If this test should be annoying for some reason, feel free to delete it as long
+// as the other test remains.
```

Comment:
Given that we have the E2E test, what's the upside of this test?

I've confirmed it does what it says on the tin, but it's quite a lot mocking / random new code to test something we're already testing.

Is the hope that we might want to test more DB logic like this in the future without it being coupled to the admin?